modules/serverless-typescript-context-object-reference.adoc

```diff
@@ -0,0 +1,130 @@
+  context.log.info(“Processing customer”);
+  return 'OK';
+}
+----
+
+You can access the function by using the `kn func emit` command to invoke it:
+
+.Example command
+[source,terminal]
+----
+$ kn func emit --sink local
+----
+
+.Example output
+[source,terminal]
+----
+{"level":30,"time":1604511655265,"pid":3430203,"hostname":"localhost.localdomain","reqId":1,"msg":"Processing customer"}
+----
+
+You can change the log level to one of `fatal`, `error`, `warn`, `info`, `debug`, `trace` or `silent`. To do that, change the value of `logLevel` to one of these values using the `config` command. You can also temporarily override the value of `logLevel` by assigning one of the log level values to the environment variable `FUNC_LOG_LEVEL`.
+
+[id="serverless-typescript-context-object-reference-query_{context}"]
+== query
+
+Returns the query string for the request, if any, as key-value pairs. These attributes are also found on the context object itself.
+
+.Example query
+[source,javascript]
+----
+function handle(context: Context) Record<string,string> {
+  // Log the 'name' query parameter
+  context.log.info(context.query.name);
+  // Query parameters are also attached to the context
+  context.log.info(context.name);
+}
+----
+
+You can access the function by using the `kn func emit` command to invoke it:
+
+.Example command
+[source,terminal]
+----
+$ kn func emit --sink local --data '{"name": "tiger"}'
+----
+
+.Example output
+[source,terminal]
+----
+{"level":30,"time":1604511655265,"pid":3430203,"hostname":"localhost.localdomain","reqId":1,"msg":"tiger"}
+{"level":30,"time":1604511655265,"pid":3430203,"hostname":"localhost.localdomain","reqId":1,"msg":"tiger"}
+----
+
+[id="serverless-typescript-context-object-reference-body_{context}"]
+== body
+
+Returns the request body if any. If the request body contains JSON code, this will be parsed so that the attributes are directly available.
+
+.Example body
+[source,javascript]
+----
+function handle(context: Context): string {
+  // log the incoming request body's 'hello' parameter
+  context.log.info(context.body.hello);
+  return 'OK';
+}
+----
+
+You can access the function by using the `kn func emit` command to invoke it:
+
+.Example command
+[source,terminal]
+----
+$ kn func emit --sink local --data '{"hello": "world"}'
+----
+
+.Example output
+[source,terminal]
+----
+{"level":30,"time":1604511655265,"pid":3430203,"hostname":"localhost.localdomain","reqId":1,"msg":"world"}
+----
+
+[id="serverless-typescript-context-object-reference-headers_{context}"]
+== headers
+
+Returns the HTTP request headers as an object.
+
+.Example header
+[source,javascript]
+----
+function handle(context: Context): string {
+  context.log.info(context.headers['custom-header']);
+  return 'OK';
+}
+----
+
+You can access the function by using the `curl` command to invoke it:
+
+.Example command
+[source,terminal]
+----
+$ curl -H'x-custom-header: some-value’' http://example.com
```

Comment:
Maybe an oversight?

Reply:
@lance I couldn't figure out how to do `kn func emit` with a custom header, can you provide the invocation?
I was looking here:
https://deploy-preview-35587--osdocs.netlify.app/openshift-enterprise/latest/serverless/cli_tools/kn-func-ref.html#serverless-kn-func-emit_kn-func-ref

Reply:
Oh right - I wasn't paying attention, just noticed the `curl`. You cannot do that at the moment. Carry on!

Reply:
:+1: